src/ResourceManager/Websites/Commands.Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs

```diff
@@ -90,7 +90,7 @@ public override void ExecuteCmdlet()
 
             var webApp = WebsitesClient.GetWebApp(ResourceGroupName, Name, null);
 
-            WriteObject(WebsitesClient.CreateWebApp(ResourceGroupName, Name, Slot, webApp.Location, AppServicePlan, cloningInfo, AseName, AseResourceGroupName));
+            WriteObject(WebsitesClient.CreateWebApp(ResourceGroupName, Name, Slot, webApp.Location, AppServicePlan==null?webApp.ServerFarmId:AppServicePlan, cloningInfo, AseName, AseResourceGroupName));
```

Comment:
why not AppServicePlan ?? webApp.ServerFarmId

Reply:
AppServicePlan  is the name of the Appservice plan where ServerFarmId is the completeId. The AppServicePlan is not exposed via the swagger on the siteObject, only serverfarmId is since this what we need in the API. If you look at the createWebApp call looks like its expecting the serverfarmId, however i verified that passing the AppServicePlan alone is accepted as well.
 createdWebSite = WrappedWebsitesClient.WebApps.CreateOrUpdateSlot(
                        resourceGroupName, webAppName, slot: slotName, siteEnvelope:
                        new Site
                        {
                            Location = location,
                            ServerFarmId = serverFarmId,
                            CloningInfo = cloningInfo,
                            HostingEnvironmentProfile = profile
                        });